### PR TITLE
Update JobResult manager store_result method to censor sensitive vari…

### PR DESCRIPTION
**Pull request by HanlinMiao**
…ables

# Closes: #3692
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/managers.py [hunk 1]
```diff
@@ -86,6 +86,11 @@ def store_result(
             "worker": worker,
         }
 
+        job = Job.objects.get(name=task_name)
```
Comment (gsnider2195) [inline nautobot/extras/managers.py:89]:
```suggestion
        job = Job.objects.get(id=job_model_id)
```

File: nautobot/extras/managers.py [hunk 2]
```diff
@@ -86,6 +86,11 @@ def store_result(
...
+        if job.has_sensitive_variables:
+            fields["task_args"] = ""
+            fields["task_kwargs"] = ""
```
Comment (gsnider2195) [inline nautobot/extras/managers.py:92]: These are JSON fields so you probably need to just delete them from `fields`

Comment (jathanism) [reply nautobot/extras/managers.py:92]: Yeah I'd `del` them.

**Comment by HanlinMiao** [comment]
`create_job_result_and_run_job` function has an `*args` but we never use it? Is that how you pass in `task_args` in a testing job?

**Comment by gsnider2195** [comment]
> `create_job_result_and_run_job` function has an `*args` but we never use it? Is that how you pass in `task_args` in a testing job?

It is but we currently don't use positional args for jobs and I think the plan was to discourage their use.

File: changes/3692.changed [hunk 3]
```diff
@@ -0,0 +1 @@
+Update `JobResult` manager `store_result` method to censor sensitive variables.
```
Comment (gsnider2195) [inline changes/3692.changed:1]:
```suggestion
Updated `JobResult` manager `store_result` method to censor sensitive variables.
```

File: nautobot/extras/tests/test_jobs.py [hunk 4]
```diff
@@ -77,6 +77,18 @@ def test_job_pass(self):
...
         self.assertEqual(job_result.status, JobResultStatusChoices.STATUS_SUCCESS)
 
+    def test_job_result_manager_censor_sensitive_variables(self):
+        """
+        Job test with JobResult Censored Sensitive Variables.
+        """
+        module = "test_has_sensitive_variables"
+        name = "TestHasSensitiveVariables"
+        # This function create_job_result_and_run_job and the subsequent functions' arguments are very messy
+        job_result = create_job_result_and_run_job(module, name, "local", 1, 2, "3", kwarg_1=1, kwarg_2="2")
```
Comment (gsnider2195) [inline nautobot/extras/tests/test_jobs.py:87]: Your run method signature for `TestHasSensitiveVariables` needs to match the args/kwargs here

Comment (HanlinMiao) [reply nautobot/extras/tests/test_jobs.py:87]: Like this correct? https://github.com/nautobot/nautobot/pull/3800/commits/9ed4cd879272f922f353a6e284ab68bd9bbd6305

File: nautobot/extras/tests/example_jobs/test_has_sensitive_variables.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    class Meta:
+        has_sensitive_variables = True
+
+    def run(self, *args, **kwargs):
+        """
+        Job function.
+        """
+        self.log_success(obj=None)
+        self.status = "complete"
```
Comment (gsnider2195) [inline nautobot/extras/tests/example_jobs/test_has_sensitive_variables.py:20]:
```suggestion
```

The status is tracked on the job result and is automatic now

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**